pi_ldapproxy/appcache.py

```diff
@@ -1,24 +1,43 @@
+import functools
+
 from twisted.internet import reactor
 from twisted.logger import Logger
 
 log = Logger()
 
+def case_insensitive_dn(wrapped_function):
+    """
+    This decorator is used in the ``AppCache`` class to implement case-insensitive DN storage.
+    :param wrapped_function: A function accepting ``dn`` as first argument
+    :return: Wrapper function which converts the DN to lowercase before passing it to ``wrapped_function``
+    """
+    @functools.wraps(wrapped_function)
+    def dn_wrapper(self, dn, *args, **kwargs):
+        if self.case_insensitive:
```

Comment:
Is there a ``self``? Will the decorator take the ``self`` from the class of the wrapped method?

Decorators are spooky.

Reply:
Well yes, in this case it works, because ``app_cache.add_to_cache(dn)`` is kind of the same as  ``AppCache.add_to_cache(self, app, dn)`` :-)